src/oauth2-mock-server.ts

```diff
@@ -125,47 +126,44 @@ function parsePort(portStr) {
   return port;
 }
 
-function parseJWK(filename) {
+async function parseJWK(filename: string): Promise<JWK.Key> {
   const jwkStr = fs.readFileSync(filename, 'utf8');
-  return JSON.parse(jwkStr);
+  return await JWK.asKey(jwkStr);
 }
 
-function parsePEM(filename) {
-  return {
+async function parsePEM(filename: string): Promise<JWK.Key> {
+  const pem = fs.readFileSync(filename, 'utf8');
+  return await JWK.asKey(pem, 'pem', {
     kid: path.parse(filename).name,
-    pem: fs.readFileSync(filename, 'utf8'),
-  };
+  });
 }
 
-function saveJWK(keys) {
+function saveJWK(keys: JWK.Key[]) {
   keys.forEach((key) => {
     const filename = `${key.kid}.json`;
     fs.writeFileSync(filename, JSON.stringify(key.toJSON(true), null, 2));
     console.log(`JSON web key written to file "${filename}".`);
   });
 }
 
-function savePEM(keys) {
+function savePEM(keys: JWK.Key[]) {
   keys.forEach((key) => {
     const filename = `${key.kid}.pem`;
     fs.writeFileSync(filename, key.toPEM(true));
     console.log(`PEM-encoded key written to file "${filename}".`);
   });
 }
 
-async function startServer(opts) {
+async function startServer(opts: Options) {
   const server = new OAuth2Server();
 
-  await Promise.all(opts.keys.map(async (key) => {
-    let jwk;
-    if (key.pem) {
```

Comment:
This is (I believe) the only change that I made from a functional standpoint. As all stored keys are now `JWK.Key`s we seem to no longer need this. To be thoroughly reviewed.